File: services/chess_service/src/main/java/net/chess_platform/chess_service/chess/piece/Rook.java
package net.chess_platform.chess_service.chess.piece;

import java.util.List;

import net.chess_platform.chess_service.chess.Chessboard;
import net.chess_platform.chess_service.chess.move.Move;
import net.chess_platform.chess_service.chess.piece.behavior.PieceBehavior;
import net.chess_platform.chess_service.chess.piece.behavior.RookBehavior;

public class Rook extends AbstractPiece {

    private PieceBehavior rookMoves = new RookBehavior();

    public Rook(Color color) {
        super(color, Type.ROOK);
    }

    @Override
    public List<Move> getMoves(Chessboard board, int row, int col) {
        return rookMoves.getMoves(board, getType(), getColor(), row, col);
    }

    @Override
    public String toString() {
        return getColor() == Color.WHITE ? Character.toString(9814) : Character.toString(9820);
    }

}

File: app/src/chess/pieces/Rook.ts
import { Chessboard } from "../Chessboard";
import { Move } from "../moves/Move";
import { AbstractPiece } from "./AbstractPiece";
import { RookBehavior } from "./behavior/RookBehavior";
import { Color } from "./Piece";

export class Rook extends AbstractPiece {

    private behavior = new RookBehavior();

    constructor(color: Color, moveCount?: number) {
        super(color, "ROOK", moveCount);
    }

    getLegalMovesImpl(board: Chessboard, row: number, col: number): Move[] {
        return this.behavior.getMoves(board, this.getType(), this.getColor(), row, col);
    }

    override toString(): string {
        return this.getColor() == "WHITE" ? String.fromCodePoint(9814) : String.fromCodePoint(9820);
    }
}